core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1682,7 +1682,14 @@ private long createDirectoryAndJournal(LockedInodePath inodePath, CreateDirector
       CreateDirectoryOptions options) throws InvalidPathException, FileAlreadyExistsException,
       IOException, AccessControlException, FileDoesNotExistException {
     try {
-      return mInodeTree.createPath(inodePath, options);
+      InodeTree.CreatePathResult createResult = mInodeTree.createPath(inodePath, options);
+      InodeDirectory inodeDirectory = (InodeDirectory) inodePath.getInode();
+      // If inodeDirectory's ttl not equals Constants.NO_TTL, it should insert into mTtlBuckets
+      if (createResult.getCreated().size() > 0) {
```

Comment:
the unit test would do the following:

1) use manual scheduling for TTL
2) request directory creation with TTL set
3) stop block master and file system master
4) start block master and file system master again with the same journal
5) manually schedule TTL and check the directory has been freed / deleted

I am happy to add this test in a followup PR myself as the setup is non-trivial.